src/group_theory/nilpotent.lean

```diff
@@ -703,3 +703,46 @@ begin
   rw [eq_bot_iff, ←hn],
   exact derived_le_lower_central n,
 end
+
+lemma normalizer_condition_of_is_nilpotent [h : is_nilpotent G] : normalizer_condition G :=
+begin
+  -- roughly based on https://groupprops.subwiki.org/wiki/Nilpotent_implies_normalizer_condition
+  rw normalizer_condition_iff_only_full_group_self_normalizing,
+  unfreezingI { induction h using nilpotent_center_quotient_ind }; clear _inst_1 G,
+  { rintros H -,
+    apply subsingleton.elim, },
+  { unfreezingI { rename [ h_G → G , h_ih → ih ] },
```

Comment:
Is it possible to avoid mentioning the auto-generated names?

Reply:
Yes, I can pass `with` to `induction`. Will do.

Unfortunately, I still have to rename afterwards, because this `induction` can’t revert the original `G` completely…